packages/agreement-process/src/services/agreementService.ts

```diff
@@ -324,17 +577,56 @@ export function agreementServiceBuilder(
       { authData, correlationId, logger }: WithLogger<AppContext>
     ): Promise<Agreement> {
       logger.info(`Suspending agreement ${agreementId}`);
-      const [agreement, events] = await suspendAgreementLogic({
+
+      const agreement = await retrieveAgreement(agreementId, agreementQuery);
+
+      assertRequesterIsConsumerOrProducer(agreement.data, authData);
+
+      assertExpectedState(
         agreementId,
+        agreement.data.state,
+        agreementSuspendableStates
+      );
+
+      const eservice = await retrieveEService(
+        agreement.data.eserviceId,
+        eserviceQuery
+      );
+
+      const descriptor = eservice.descriptors.find(
+        (d) => d.id === agreement.data.descriptorId
+      );
+      assertDescriptorExist(
+        eservice.id,
+        agreement.data.descriptorId,
+        descriptor
+      );
+
+      const updatedAgreement: Agreement = createAgreementSuspended({
+        agreement: agreement.data,
         authData,
-        agreementQuery,
-        tenantQuery,
-        eserviceQuery,
-        correlationId,
+        descriptor,
+        consumer: await retrieveTenant(agreement.data.consumerId, tenantQuery),
       });
-      await repository.createEvent(events);
 
-      return agreement;
+      const isProducer = authData.organizationId === agreement.data.producerId;
+      const isConsumer = authData.organizationId === agreement.data.consumerId;
+
+      if (!isProducer && !isConsumer) {
+        throw new Error(
+          "Agreement can only be suspended by the consumer or producer."
+        );
+      }
+
+      const eventType = isProducer
+        ? toCreateEventAgreementSuspendedByProducer
+        : toCreateEventAgreementSuspendedByConsumer;
+
+      await repository.createEvent(
+        eventType(updatedAgreement, agreement.metadata.version, correlationId)
+      );
```

Comment:
Left a comment also in the corresponding processor file: I would keep the creation of the event in the processor, as we are doing for other processors e.g., submission, upgrade, etc. and just call the `.createEvent` on the return value here